tests/integration/test_auth_client_flow.py

```diff
@@ -0,0 +1,133 @@
+from six.moves.urllib.parse import quote_plus
+
+import globus_sdk
+from globus_sdk.auth.oauth2_constants import DEFAULT_REQUESTED_SCOPES
+from globus_sdk.auth.oauth2_native_app import make_native_app_challenge
+from tests.framework import CapturedIOTestCase, get_client_data
+from globus_sdk.exc import GlobusAPIError
+
+
+class AuthClientIntegrationTests(CapturedIOTestCase):
+
+    def test_oauth2_get_authorize_url_native(self):
+        """
+        Starts an auth flow with a NativeAppFlowManager, gets the authorize url
+        validates expected results with both default and specified parameters.
+        """
+        ac = globus_sdk.AuthClient(
+            client_id=get_client_data()["native_app_client1"]["id"])
+
+        # default parameters for starting auth flow
+        flow_manager = globus_sdk.auth.GlobusNativeAppFlowManager(ac)
+        ac.current_oauth2_flow_manager = flow_manager
+
+        # get url_and validate results
+        url_res = ac.oauth2_get_authorize_url()
+        expected_vals = [ac.base_url + "v2/oauth2/authorize?",
+                         "client_id=" + ac.client_id,
+                         "redirect_uri=" +
+                         quote_plus(ac.base_url + "v2/web/auth-code"),
+                         "scope=" + quote_plus(DEFAULT_REQUESTED_SCOPES),
+                         "state=" + "_default",
+                         "response_type=" + "code",
+                         "code_challenge=" +
+                         quote_plus(flow_manager.challenge),
+                         "code_challenge_method=" + "S256",
+                         "access_type=" + "online"]
+        for val in expected_vals:
+            self.assertIn(val, url_res)
+
+        # starting flow with specified paramaters
+        flow_manager = globus_sdk.auth.GlobusNativeAppFlowManager(
+            ac, requested_scopes="scopes", redirect_uri="uri",
+            state="state", verifier="verifier", refresh_tokens=True)
+        ac.current_oauth2_flow_manager = flow_manager
```

Comment:
I don't _object_ to us testing this, but this here is one of the side effects of the fact that the base `AuthClient` is never expected to have a flow manager. I wouldn't bother stripping out these tests, but you could have skipped the tests on `AuthClient` and it would have done no harm.